cannon/mipsevm/tests/evm_multithreaded_test.go

```diff
@@ -1187,6 +1187,48 @@ func TestEVM_UnsupportedSyscall(t *testing.T) {
 	}
 }
 
+func TestEVM_EmptyThreadStacks(t *testing.T) {
+    t.Parallel()
+    var tracer *tracing.Hooks
+
+    cases := []struct {
+        name            string
+        activeStackSize int
+        otherStackSize  int
+        shouldPanic     bool
+    }{
+        {name: "Empty stacks", activeStackSize: 0, otherStackSize: 0, shouldPanic: true},
+        {name: "Active stack only", activeStackSize: 1, otherStackSize: 0, shouldPanic: false},
+        {name: "Other stack only", activeStackSize: 0, otherStackSize: 1, shouldPanic: true},
+        {name: "Active and other stacks", activeStackSize: 1, otherStackSize: 1, shouldPanic: false},
+        {name: "Multiple active stacks", activeStackSize: 2, otherStackSize: 0, shouldPanic: false},
+        {name: "Multiple other stacks", activeStackSize: 0, otherStackSize: 2, shouldPanic: true},
+        {name: "Multiple stacks both", activeStackSize: 2, otherStackSize: 2, shouldPanic: false},
+    }
+
+    for i, c := range cases {
+        c := c
+		i := i
+        t.Run(c.name, func(t *testing.T) {
+            t.Parallel()
+            goVm, state, contracts := setup(t, i*123, nil)
+            mttestutil.SetupThreads(int64(i*123), state, false, c.activeStackSize, c.otherStackSize)
+
+            if c.shouldPanic {
+                require.Panics(t, func() {
+                    goVm.Step(true)
+                    testutil.AssertEVMReverts(t, state, contracts, tracer)
+                })
+            } else {
+                require.NotPanics(t, func() {
+                    goVm.Step(true)
+                    testutil.AssertEVMReverts(t, state, contracts, tracer)
```

Comment:
For the case where `shouldPanic == false`, you should use `testutil.ValidateEVM` as is done [here](https://github.com/ethereum-optimism/optimism/blob/bd50ac6909245e6a283e824995940fc055a90f14/cannon/mipsevm/tests/evm_common_test.go#L419). `AssertEVMReverts` doesn't actually panic. But rather runs a `require` assertion against the EVM execution status.